util/util.go

```diff
@@ -95,6 +97,16 @@ func RandInit() {
 	math_rand.Seed(int64(binary.LittleEndian.Uint64(b[:])))
 }
 
+// Generate crytographically secure challenge
+func GenChallenge() string {
+	//call on the import DUO method
+	challenge, err := protocol.CreateChallenge()
+	if err != nil {
+		panic("Failed to generate cryographically secure challenge")
+	}
+	return base64.RawURLEncoding.EncodeToString(challenge)
+}
+
 // Generate a random string of alpha characters of length n
 func RandStringBytesRmndr(n int) []byte {
```

Comment:
Hey @Oneiroi Yep looks like you can remove RandInit and RandStringBytesRmndr

Reply:
over to you @pdavies011010 🙇